test/src/loaders/workspace_uploader/workspace_uploader_test.py

```diff
@@ -1,5 +1,210 @@
+    assembly_dir, assembly_name = assembly_dirs[0], assembly_names[0]
+
+    file_path = workspace_uploader._get_source_file(assembly_dir, assembly_name)
+    expected_target_file_path = os.path.abspath(target_files[0])
+
+    assert expected_target_file_path == file_path
+
+
+def test_read_upload_status_yaml_file(setup_and_teardown):
+    _, assembly_dirs, assembly_names, _, workspace_id, upload_env_key, _ = setup_and_teardown
+    assembly_dir = assembly_dirs[0]
+    assembly_name = assembly_names[0]
+
+    # test empty yaml file in assembly_dir
+    data, uploaded = workspace_uploader._read_upload_status_yaml_file(
+        upload_env_key, workspace_id, assembly_dir, assembly_name
+    )
+    
+    expected_data = {upload_env_key: {workspace_id: dict()}}
+    
+    assert not uploaded
+    assert expected_data == data
+
+
+def test_update_upload_status_yaml_file(setup_and_teardown):
+    _, assembly_dirs, assembly_names, upas, workspace_id, upload_env_key, _ = setup_and_teardown
+    upa = upas[0]
+    assembly_dir = assembly_dirs[0]
+    assembly_name = assembly_names[0]
+    
+    workspace_uploader._update_upload_status_yaml_file(
+        upload_env_key, workspace_id, upa, assembly_dir, assembly_name
+    )
+    data, uploaded = workspace_uploader._read_upload_status_yaml_file(
+        upload_env_key, workspace_id, assembly_dir, assembly_name
+    )
+
+    expected_data = {
+        upload_env_key: {workspace_id: {"file_name": assembly_name, "upa": upa}}
+    }
+
+    assert uploaded
+    assert expected_data == data
+
+
+def test_fetch_assemblies_to_upload(setup_and_teardown):
+    tmp_dir, assembly_dirs, assembly_names, _, workspace_id, upload_env_key, _ = setup_and_teardown
+    collection_source_dir = Path(tmp_dir) / "collectionssource"
+    upload_file_ext = ["genomic.fna.gz"]
+
+    count, wait_to_upload_assemblies = workspace_uploader._fetch_assemblies_to_upload(
+        upload_env_key, workspace_id, collection_source_dir, upload_file_ext
+    )
+    
+    expected_count = len(assembly_names)
+    expected_wait_to_upload_assemblies = {
+        assembly_name: assembly_dir
+        for assembly_name, assembly_dir in zip(assembly_names, assembly_dirs)
+    }
+
+    assert expected_count == count
+    assert expected_wait_to_upload_assemblies == wait_to_upload_assemblies
+
+
+def test_prepare_skd_job_dir_to_upload(setup_and_teardown):
+    tmp_dir, assembly_dirs, assembly_names, _, _, _, _ = setup_and_teardown
+    wait_to_upload_assemblies = {
+        assembly_name: assembly_dir
+        for assembly_name, assembly_dir in zip(assembly_names, assembly_dirs)
+    }
+
+    conf = Mock()
+    username = "kbase"
+    job_dir = loader_helper.make_job_dir(tmp_dir, username)
+    conf.job_data_dir = loader_helper.make_job_data_dir(job_dir)
+    data_dir = workspace_uploader._prepare_skd_job_dir_to_upload(conf, wait_to_upload_assemblies)
+    
+    assert sorted(os.listdir(data_dir)) == sorted(assembly_names)
```

Comment:
Should check the hardlinks were made as expected

Reply:
Updated!

Reply:
:+1: